admin/app/components/solidus_admin/main_nav_item/component.rb

```diff
@@ -5,16 +5,13 @@ module SolidusAdmin
   class MainNavItem::Component < BaseComponent
     with_collection_parameter :item
 
-    attr_reader :item
```

Comment:
A random note about using `ivar` or `attr_reader`: `ivar` is harder to deprecate against the `attr_reader`. For the former, we will need to use [DeprecatedInstanceVariableProxy](https://api.rubyonrails.org/classes/ActiveSupport/Deprecation/DeprecatedInstanceVariableProxy.html) which is complex and doesn't support deprecating without a replacement if I recall correctly. The `attr_reader` is easier to deprecate, because it's a plain method.

Reply:
That's a good point, but if we allow the specific granularity in customizing only the component template without touching the rest of the component, and the template relies on that instance variable, it's already a public API that people might be using in a legit way.

Reply:
@kennyadsl yes, that's something we should seriously consider before committing to allowing replacing just the template